src/shacl-shape-lib/connectors-shacl-shape.xsl

```diff
@@ -165,33 +281,36 @@
 
         <xsl:variable name="targetClassURI"
             select="f:buildURIfromLexicalQName($connector/target/model/@name)"/>
+        <xsl:variable name="sourceClassURI"
+            select="f:buildURIfromLexicalQName($connector/source/model/@name)"/>
+        <xsl:variable name="normalisedSourceClassName"
+            select="replace($connector/source/model/@name, ':', '-')"/>
         <xsl:variable name="targetRole"
             select="
                 if (boolean($connector/target/role/@name)) then
                     $connector/target/role/@name
                 else
                     fn:error(xs:QName('connectors'),concat($connector/@xmi:idref, ' - connector target role name is empty'))"/>
+        <xsl:variable name="normalisedTargetRole" select="replace($targetRole, ':', '-')"/>
         <xsl:variable name="targetRoleURI"
             select="f:buildURIfromLexicalQName($targetRole)"/>
         <xsl:variable name="connectorDirection" select="$connector/properties/@direction"/>
 
         <xsl:if test="$connectorDirection = 'Source -&gt; Destination'">
 
-            <sh:NodeShape rdf:about="{$targetRoleURI}">
-                <sh:property>
-                    <sh:PropertyShape>
-                        <sh:path rdf:resource="http://www.w3.org/2004/02/skos/core#inScheme"/>
-                        <sh:hasValue rdf:resource="{$targetClassURI}"/>
-                    </sh:PropertyShape>
-                </sh:property>
-            </sh:NodeShape>
+
+            <rdf:Description
+                rdf:about="{fn:concat($base-shape-uri,$defaultDelimiter,$normalisedSourceClassName,'-',$normalisedTargetRole)}">
```

Comment:
here you don't even create a variable but concat directly ?! We can be more professional than this.